crypto/engine/eng_list.c

```diff
@@ -278,6 +360,8 @@ static void engine_cpy(ENGINE *dest, const ENGINE *src)
     dest->load_pubkey = src->load_pubkey;
     dest->cmd_defns = src->cmd_defns;
     dest->flags = src->flags;
+    dest->dynamic_id = src->dynamic_id;
+    engine_add_dynamic_id(dest, NULL);
```

Comment:
oops, here the global lock is already held, so I will have to add a not_locked parameter to this function